coordinator/internal/authority/authority.go

```diff
@@ -181,52 +189,145 @@ func (m *Authority) GetCertBundle(peerPublicKeyHashStr string) (Bundle, error) {
 }
 
 // GetManifestsAndLatestCA retrieves the manifest history and the currently active CA instance.
-func (m *Authority) GetManifestsAndLatestCA() ([]*manifest.Manifest, *ca.CA) {
-	// TODO(burgerdev): The CA should be tied to the manifest.
-	return m.manifests.All(), m.ca.Load()
+//
+// If no manifest is configured, it returns an empty slice and a nil CA.
+func (m *Authority) GetManifestsAndLatestCA() ([]*manifest.Manifest, *ca.CA, error) {
+	if err := m.sync(); err != nil {
+		return nil, nil, fmt.Errorf("could not sync internal state: %w", err)
+	}
+	state := m.state.Load()
+	if state == nil {
+		return nil, nil, nil
```

Comment:
returning nil, nil (,nil) is usually discouraged.

Reply:
Would you rather fail the call, or return an empty slice and nil CA?

Reply:
We are returning an error on the caller side anyway if we get 3x nil, right? So we might want to raise it here.

Reply:
Added a test for it that revealed a bug :)